api/src/main/java/org/openmrs/api/TagService.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api;
+
+import org.openmrs.OpenmrsObject;
+import org.openmrs.TagConstants;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.api.OpenmrsService;
+import org.openmrs.Tag;
+import org.openmrs.api.context.Context;
+
+import java.util.List;
+
+/**
+ * Contains services for adding/removing/obtaining Tags. Example Usage: <br>
+ * <code>
+ *   List&lt;Tag&gt; tags = Context.getService(TagService.class).getAllTags();
+ * </code>
+ * 
+ * @see org.openmrs.Tag
+ */
+
+public interface TagService extends OpenmrsService {
+	
+	/**
+	 * Fetches a tag by uuid. It can be called by any authenticated user.
+	 * 
+	 * @param uuid Universally Unique Identifier for the given tag
+	 * @return tag or null
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	Tag getTagByUuid(String uuid);
+	
+	/**
+	 * Saves a tag. Tags can only be added not edited.
+	 * 
+	 * @param tag the Tag Object to be saved
+	 * @return Tag that has been saved
+	 */
+	@Authorized(TagConstants.MANAGE_TAGS)
+	Tag saveTag(Tag tag);
+	
+	/**
+	 * Completely delete the tag from the database.
+	 * 
+	 * @param tag The Tag to remove from the system
+	 */
+	@Authorized(TagConstants.MANAGE_TAGS)
+	void purgeTag(Tag tag);
+	
+	/**
+	 * Removes a tag(if exists) from an OpenmrsObject.
+	 * 
+	 * @param openmrsObject the openmrsObject from which the tags is to be removed
+	 * @param tag the textual tag to be removed
+	 */
+	@Authorized(TagConstants.MANAGE_TAGS)
+	void removeTag(OpenmrsObject openmrsObject, String tag);
```

Comment:
This should return a boolean, true if the tag was found and removed otherwise false